libraries/chain/asset_evaluator.cpp

```diff
@@ -378,13 +378,16 @@ void check_children_of_bitasset(database& d, const asset_update_bitasset_operati
       return;
 
    // loop through all assets that have this asset as a backing asset
-   const auto& idx = d.get_index_type<asset_index>().indices().get<by_type>();
-
-   for( auto itr = idx.lower_bound(true); itr != idx.end(); ++itr )
+   const auto& idx = d.get_index_type<graphene::chain::asset_bitasset_data_index>().indices().get<by_short_backing_asset>();
+   auto backed_by_itr = idx.find( op.asset_to_update );
+   if ( backed_by_itr != idx.end() ) // if we found at least 1
```

Comment:
I think this if is not needed.

Reply:
Unfortunately it is needed. In index [ 2, 4, 6, 8 ] ``upper_bound(5)`` points to 6, ``find(5)`` points to 8, so ``upper_bound != find``, and the ``for`` below enters when it is not supposed to, and possibly (gulp!) never ends.

Reply:
How about using https://www.boost.org/doc/libs/1_55_0/libs/multi_index/doc/reference/ord_indices.html#range_operations instead?

Reply:
Switched to using range. Take a look. The lambda syntax there looks hard to read IMO, but maybe it is just me. Had to include an extra boost header that throws deprecation warnings, but it is necessary for boost <= 1.63.

I am unaware of find() not working as expected on non-unique indexes. If you could point me to anything on the subject I'd appreciate it. If nothing is handy, I'll just keep it in mind until I run across something that proves or disproves it.

Reply:
I'd rather code like this
```
auto itr = idx.lower_bound( xxx );
auto end = idx.upper_bound( xxx );
while( itr != end )
{
   process();
   ++itr;
}
```

If using `range` something (although I don't like lambda)
```
auto range = idx.equal_range( xxx );
std::for_each(range.first, range.second,
                    [&something](const yyy_obj& zzz) {
                       process();
                    });
```

Reply:
I think that looks a lot cleaner. Thanks @abitmore and @pmconrad. Testing now, should be checked in soon.

Reply:
> I am unaware of find() not working as expected on non-unique indexes.

An ordered non-unique boost multi-index is equivalent to a std::multiset. Look at the definition of std::multiset::find - it "Finds an element with key equivalent to key", but doesn't make any other guarantees.